lib/cloudfront_signer/policy.ex

```diff
@@ -4,11 +4,68 @@ defmodule CloudfrontSigner.Policy do
   """
   defstruct [:resource, :expiry]
 
-  @type t :: %__MODULE__{}
+  @type t :: %__MODULE__{
+          resource: String.t(),
+          expiry: integer()
+        }
+
+  @typedoc """
+  Type definition for an RSA private key in ASN.1 format.
+  The tuple elements represent:
+  - :RSAPrivateKey - The key type identifier
+  - :"two-prime" - The RSA version identifier
+  - modulus - The RSA modulus n
+  - publicExponent - The public exponent e
+  - privateExponent - The private exponent d
+  - prime1 - The first prime factor p of n
+  - prime2 - The second prime factor q of n
+  - exponent1 - d mod (p-1)
+  - exponent2 - d mod (q-1)
+  - coefficient - The CRT coefficient q^(-1) mod p
+  - otherPrimeInfos - Optional additional prime info (usually :asn1_NOVALUE)
+  """
+  @type rsa_private_key :: {
+          :RSAPrivateKey,
+          :"two-prime",
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          :asn1_NOVALUE | [non_neg_integer()]
+        }
```

Comment:
There is an rsa_private_key type in :public_key, but dialyzer couldn't find it when running in Github. I tried adding :public_key to the plt_add_apps, but that didn't solve the problem.

Interestingly, dialyzer did not emit a warning locally.

https://www.erlang.org/doc/apps/public_key/public_key.html